lib/msal-core/src/utils/UrlUtils.ts

```diff
@@ -244,4 +245,27 @@ export class UrlUtils {
         const hash = UrlUtils.getHashFromUrl(urlFragment);
         return CryptoUtils.deserialize(hash);
     }
+
+    /**
+     * @hidden
+     * generate unique state per request
+     * @param request
+     */
+    static getRequestState(state: string): string {
```

Comment:
These aren't URL specific functions, so i dont think they should go here. I know we removed this from ServerrequestParams, but I feel that is the most logical place for it. We should either make a request Utils function or find another design.

Reply:
Following this up with a requestValidation utility function in a new PR.